MovieDirectory/MovieDirectory/Strategies/IActionMovie.cs

```diff
@@ -0,0 +1,14 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using DataMovie;
+
+namespace MovieDirectory
+{ 
+    public interface IActionMovie<T>
+    {
+        public void Action(Dictionary<int, Movie> movies, params T[] data);
```

Comment:
I think you can make it's just `T data` and for each operation create a separate model.
For example `CreateMovieModel` with specific fields. This way you don't operate with arrays, which is not safe and clean